### PR TITLE
sql/migrate: increase error verbosity

### PR DESCRIPTION
**Pull request by masseelch**

### CODE REVIEW

File: sql/migrate/lex.go [hunk 1]
```diff
@@ -174,10 +174,15 @@ func (l *lex) addPos(p int) {
 }
 
 func (l *lex) skipQuote(quote rune) error {
+	pos := l.pos
 	for {
 		switch r := l.next(); {
 		case r == eos:
-			return fmt.Errorf("unclosed quote %q", quote)
+			end := pos + 15
```
Comment (giautm) [inline sql/migrate/lex.go:181]: Why this number? Can you please put some comment?

Comment (masseelch) [reply sql/migrate/lex.go:181]: Arbitrary number to provide some more feedback to where the unclosed quote is

What do you think is a reasonable length?

Comment (rotemtam) [reply sql/migrate/lex.go:181]: From the tests it looks good . Just put it in a const with some descriptive name with some comment

Comment (a8m) [reply sql/migrate/lex.go:181]: Position is not that good. Change it to line number + column number. Then, adding this string prefix is not needed.

File: sql/migrate/lex.go [hunk 2]
```diff
@@ -174,10 +174,21 @@ func (l *lex) addPos(p int) {
...
+				s    = l.input[:pos]
+				col  = strings.LastIndex(s, `\n`)
+				line = 1 + strings.Count(s, `\n`)
+			)
+			if line == 1 {
+				col = pos
+			} else {
+				col = pos - col - 2
+			}
+			return fmt.Errorf("unclosed quote %q at %d:%d", quote, line, col)
```
Comment (a8m) [inline sql/migrate/lex.go:191]: Maybe add an `error/f` method and report this format for all errors:

```go
file:%d:%d: <message>
```
`file` can be constant or omitted as we don't have this info.

File: sql/migrate/lex.go [hunk 3]
```diff
@@ -281,3 +282,18 @@ func (l *lex) setDelim(d string) error {
 	l.delim = strings.NewReplacer(`\n`, "\n", `\r`, "\r", `\t`, "\t").Replace(d)
 	return nil
 }
+
+func (l *lex) error(pos int, format string, args ...any) error {
+	format = "%d:%d: " + format
+	var (
+		s    = l.input[:pos]
+		col  = strings.LastIndex(s, `\n`)
+		line = 1 + strings.Count(s, `\n`)
```
Comment (a8m) [inline sql/migrate/lex.go:291]:
```suggestion
		col  = strings.LastIndex(s, "\n")
		line = 1 + strings.Count(s, "\n")
```

File: sql/migrate/lex_test.go [hunk 4]
```diff
@@ -105,3 +105,57 @@ cmd7;
...
+			err:  `1:1: unclosed quote '"'`,
+		},
+		{
+			name: "unclosed double at 2:2",
+			stmt: `unclosed double at 2:2\n "`,
+			err:  `2:2: unclosed quote '"'`,
+		},
+		{
+			name: "unclosed double at 5:5",
+			stmt: `unclosed double at 2:2\n\n\n\n    "`,
```
Comment (a8m) [inline sql/migrate/lex_test.go:140]:
```suggestion
			stmt: "unclosed double at 2:2\n\n\n\n    "",
```
\n inside raw strings are escaped

**Review by a8m — APPROVED** [review]
💯

**Pull request merged**